# Dataset & Data Understanding

### Load Dataset
The dataset is loaded using Pandas for further analysis and preprocessing.

In [3]:
import pandas as pd

In [4]:
df = pd.read_csv("../data/raw/student_placement.csv")

In [5]:
df.shape

(100000, 18)

In [6]:
df.head()

,branch,college_tier,cgpa,backlogs,coding_skills,dsa_score,aptitude_score,communication_skills,ml_knowledge,system_design,internships,projects_count,certifications,hackathons,open_source_contributions,extracurriculars,placement_status,salary_package_lpa
0,ECE,Tier-3,6.70,0,7.6,4.4,49.5,3.7,6.4,0.3,1,4,4,3,2,1,1,14.75
1,Chemical,Tier-2,5.70,0,5.4,7.9,72.0,8.3,6.3,1.9,0,4,0,0,0,0,0,NaN
2,EE,Tier-2,7.19,0,5.6,6.8,79.1,7.4,4.4,5.2,1,3,2,1,2,0,1,19.06
3,CE,Tier-2,6.48,0,5.2,3.1,48.4,5.0,1.1,6.7,1,4,3,0,0,0,0,NaN
4,CSE,Tier-2,6.71,1,5.9,4.7,61.2,4.3,2.7,2.8,1,2,0,3,0,1,1,13.42


In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 18 columns):
 #   Column                     Non-Null Count   Dtype  
---  ------                     --------------   -----  
 0   branch                     100000 non-null  str    
 1   college_tier               100000 non-null  str    
 2   cgpa                       100000 non-null  float64
 3   backlogs                   100000 non-null  int64  
 4   coding_skills              100000 non-null  float64
 5   dsa_score                  100000 non-null  float64
 6   aptitude_score             100000 non-null  float64
 7   communication_skills       100000 non-null  float64
 8   ml_knowledge               100000 non-null  float64
 9   system_design              100000 non-null  float64
 10  internships                100000 non-null  int64  
 11  projects_count             100000 non-null  int64  
 12  certifications             100000 non-null  int64  
 13  hackathons                 100000 non-nul

## Analysis: Dataset Overview

### Observation:

- The dataset contains **100,000 student records and 18 columns**.
- The dataset includes academic performance, technical skills, internships, projects, certifications, and placement-related information.
- The dataset contains categorical, numerical, and binary target variables.
- All columns contain **100,000 non-null values except `salary_package_lpa`**, which contains missing values for some students.

In [8]:
df.isnull().sum()

branch                           0
college_tier                     0
cgpa                             0
backlogs                         0
coding_skills                    0
dsa_score                        0
aptitude_score                   0
communication_skills             0
ml_knowledge                     0
system_design                    0
internships                      0
projects_count                   0
certifications                   0
hackathons                       0
open_source_contributions        0
extracurriculars                 0
placement_status                 0
salary_package_lpa           31525
dtype: int64

In [9]:
pd.crosstab(
    df["placement_status"],
    df["salary_package_lpa"].isna()
)

salary_package_lpa,False,True
placement_status,,
0,0,31525
1,68475,0


## Analysis: Missing Values

### Observation:

- All predictor variables contain **no missing values**.
- The `salary_package_lpa` column contains **31,525 missing values**.
- All missing salary values belong to students with `placement_status = 0`.
- This indicates that salary information is not applicable to students who were not placed.

### Business Insight:

The dataset provides complete information for the student features used for placement prediction. Salary is only relevant for placed students, so the missing salary values can be retained rather than treated as random missing data.

In [10]:
df["placement_status"].value_counts()

placement_status
1    68475
0    31525
Name: count, dtype: int64

## Analysis: Placement Distribution

### Observation:

- **68,475 students (68.48%)** are placed.
- **31,525 students (31.53%)** are not placed.
- The target variable `placement_status` contains two classes: **0 and 1**.
- The dataset contains more placed students than unplaced students.

### Business Insight:

The placement distribution provides a useful basis for building a binary classification model to predict whether a student is likely to be placed.

In [11]:
print("Branches:")
print(df["branch"].value_counts())

print("\nCollege Tiers:")
print(df["college_tier"].value_counts())

Branches:
branch
CSE         25046
IT          16065
ECE         14939
EE          12092
ME          12008
CE          10024
Chemical     9826
Name: count, dtype: int64

College Tiers:
college_tier
Tier-3    45222
Tier-2    39910
Tier-1    14868
Name: count, dtype: int64


## Analysis: Branch and College Tier Distribution

### Observation:

- The dataset contains **7 engineering branches**.
- CSE has the highest representation with **25,046 students**.
- Chemical has the lowest representation with **9,826 students**.
- The dataset contains **3 college tiers: Tier-1, Tier-2, and Tier-3**.
- Tier-3 has the highest representation with **45,222 students**.
- Tier-1 has the lowest representation with **14,868 students**.

### Business Insight:

The dataset represents students from multiple engineering branches and college tiers, allowing the model to consider academic background and college tier when predicting placement outcomes.

In [12]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
cgpa,100000.0,7.206381,0.925235,4.00,6.58,7.21,7.83,10.00
backlogs,100000.0,0.547010,0.862727,0.00,0.00,0.00,1.00,3.00
coding_skills,100000.0,5.995147,1.496302,1.00,5.00,6.00,7.00,10.00
dsa_score,100000.0,5.500711,1.781961,1.00,4.30,5.50,6.70,10.00
aptitude_score,100000.0,64.990511,11.990892,20.00,56.80,65.00,73.10,100.00
communication_skills,100000.0,5.990614,1.496552,1.00,5.00,6.00,7.00,10.00
ml_knowledge,100000.0,4.508752,1.968278,0.00,3.20,4.50,5.90,10.00
system_design,100000.0,4.008210,1.778405,0.00,2.80,4.00,5.20,10.00
internships,100000.0,1.095040,0.940264,0.00,0.00,1.00,2.00,3.00
projects_count,100000.0,2.397280,1.242829,0.00,1.00,2.00,3.00,5.00


In [14]:
print("Branch:")
print(df["branch"].unique())

print("\nCollege Tier:")
print(df["college_tier"].unique())

print("\nPlacement Status:")
print(df["placement_status"].unique())

Branch:
<ArrowStringArray>
['ECE', 'Chemical', 'EE', 'CE', 'CSE', 'IT', 'ME']
Length: 7, dtype: str

College Tier:
<ArrowStringArray>
['Tier-3', 'Tier-2', 'Tier-1']
Length: 3, dtype: str

Placement Status:
[1 0]


## Analysis: Categorical Variables

### Observation:

- The `branch` column contains **7 valid categories**: CSE, IT, ECE, EE, ME, CE, and Chemical.
- The `college_tier` column contains **3 valid categories**: Tier-1, Tier-2, and Tier-3.
- The `placement_status` column contains only **0 and 1**, making it a binary target variable.
- No unexpected categories were found in these columns.

### Business Insight:

The categorical variables provide important information about a student's academic background and college environment, which can be used during feature preprocessing and model development.

In [15]:
df.select_dtypes(include="number").agg(["min", "max"]).T

,min,max
cgpa,4.00,10.00
backlogs,0.00,3.00
coding_skills,1.00,10.00
dsa_score,1.00,10.00
aptitude_score,20.00,100.00
communication_skills,1.00,10.00
ml_knowledge,0.00,10.00
system_design,0.00,10.00
internships,0.00,3.00
projects_count,0.00,5.00


## Analysis: Numerical Feature Ranges

### Observation:

- The numerical features have values within their expected ranges.
- `cgpa` ranges from **4.00 to 10.00**.
- Skill-related features such as `coding_skills`, `dsa_score`, and `communication_skills` range from **1.00 to 10.00**.
- `aptitude_score` ranges from **20.00 to 100.00**.
- `internships`, `projects_count`, `certifications`, `hackathons`, `open_source_contributions`, and `extracurriculars` have reasonable count-based ranges.
- `placement_status` contains values from **0 to 1**, confirming that it is a binary target variable.
- No obviously invalid numerical ranges were observed.

In [16]:
df.duplicated().sum()

np.int64(0)

## Analysis: Duplicate Records

### Observation:

- The dataset contains **0 duplicate records**.
- Each student record is unique in the dataset.
- No duplicate rows need to be removed during data cleaning.

## Phase 2 Conclusion

The dataset is suitable for developing an **AI-Based Student Placement Prediction and Career Recommendation System**.
The dataset is now understood and ready for the **Data Cleaning & Preprocessing** stage.

## Analysis: Final Dataset Summary

### Observation:

- The dataset contains **100,000 student records and 18 columns**.
- It includes academic, technical, internship, project, certification, and extracurricular features related to student placement.
- The dataset contains **7 engineering branches** and **3 college tiers**.
- All predictor variables are complete, with **no missing values**.
- The only missing values are in `salary_package_lpa`, where **31,525 values are missing for unplaced students**.
- The target variable `placement_status` is binary, with **68,475 placed students and 31,525 unplaced students**.
- The numerical features have reasonable value ranges, with no obvious invalid values identified.
- The dataset contains **no duplicate records**.
- Overall, the dataset is suitable for proceeding to the **data cleaning and preprocessing stage**.

### Business Insight:

The dataset provides a large and diverse set of student-related attributes that can be used to analyze factors associated with placement outcomes. The absence of duplicate records and missing values in the predictor variables provides a good foundation for developing a reliable placement prediction model.